axelrod/strategy_transformers.py

```diff
@@ -84,24 +95,25 @@ def __call__(self, PlayerClass):
             except KeyError:
                 pass
 
-            # Is the original strategy method a static method?
-            signature = inspect.signature(PlayerClass.strategy)
-            strategy_args = [p.name for p in signature.parameters.values()
-                    if p.kind == inspect.Parameter.POSITIONAL_OR_KEYWORD]
-            is_static = True
-            if len(strategy_args) > 1:
-                is_static = False
-
             # Define the new strategy method, wrapping the existing method
             # with `strategy_wrapper`
             def strategy(self, opponent):
 
-                if is_static:
-                    # static method
+                if strategy_wrapper == dual_wrapper:
+                    # dual_wrapper figures out strategy as if the Player had
+                    # played the opposite actions of its current history.
+                    flip_play_attributes(self)
+
+                if is_strategy_static(PlayerClass):
                     proposed_action = PlayerClass.strategy(opponent)
                 else:
                     proposed_action = PlayerClass.strategy(self, opponent)
 
+                if strategy_wrapper == dual_wrapper:
+                    # after dual_wrapper_figures calls the strategy, it returns
```

Comment:
typo here - "dual_wrapper_figures" should be "dual_wrapper".  i'll wait on a full review before pushing this change.